presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -346,15 +347,39 @@ public static Optional<CompressionCodec> getCompressionCodec(TextInputFormat inp
         return Optional.ofNullable(compressionCodecFactory.getCodec(file));
     }
 
-    public static InputFormat<?, ?> getInputFormat(Configuration configuration, String inputFormatName, boolean symlinkTarget)
+    public static InputFormat<?, ?> getInputFormat(Configuration configuration, String inputFormatName, String serDe, boolean symlinkTarget)
     {
         try {
             JobConf jobConf = toJobConf(configuration);
 
             Class<? extends InputFormat<?, ?>> inputFormatClass = getInputFormatClass(jobConf, inputFormatName);
             if (symlinkTarget && (inputFormatClass == SymlinkTextInputFormat.class)) {
-                // symlink targets are always TextInputFormat
-                inputFormatClass = TextInputFormat.class;
+                if (serDe == null) {
+                    throw new PrestoException(HIVE_UNSUPPORTED_FORMAT, "Missing SerDe for SymlinkTextInputFormat");
+                }
+
+                /*
+                 * https://github.com/apache/hive/blob/b240eb3266d4736424678d6c71c3c6f6a6fdbf38/ql/src/java/org/apache/hadoop/hive/ql/io/SymlinkTextInputFormat.java#L47-L52
+                 * According to Hive implementation of SymlinkInputFormat, The target input data should be in TextInputFormat.
+                 *
+                 * But Delta Lake provides an integration with Presto using Symlink Tables with target input data as MapredParquetInputFormat.
+                 * https://docs.delta.io/latest/presto-integration.html
+                 *
+                 * To comply with Hive implementation, we will keep the default value here as TextInputFormat unless serde is not LazySimpleSerDe
+                 */
+                if (serDe.equals(TEXTFILE.getSerDe())) {
+                    inputFormatClass = TextInputFormat.class;
+                    return ReflectionUtils.newInstance(inputFormatClass, jobConf);
+                }
```

Comment:
While writing the unit tests, I observed that both `SEQUENCEFILE` and `TEXTFILE` use the same `LazySimpleSerDe`. As a result, the method was always returning `SEQUENCEFILE` since it appears before `TEXTFILE` in the `HiveStorageFormat` enum. However, according to the Hive specification, Symlink Tables point to data with `TextInputFormat`, so I’ve updated the mapping for `LazySimpleSerDe` to return `TEXTFILE`.